### PR TITLE
Feature energy

### PR DESCRIPTION
**Pull request by jmvegh**
updated way we represented energy; regression scripts now pass

### CODE REVIEW

File: scripts/experimental/PYOPTimization/SNOPT_print.out [hunk 1]
```diff
@@ -0,0 +1,89 @@
+
```
Comment (aerialhedgehog) [inline scripts/experimental/PYOPTimization/SNOPT_print.out]: delete this file

File: scripts/experimental/PYOPTimization/SNOPT_summary.out [hunk 2]
```diff
@@ -0,0 +1,21 @@
+
```
Comment (aerialhedgehog) [inline scripts/experimental/PYOPTimization/SNOPT_summary.out]: delete this file

File: scripts/regression/test_mission_Embraer_E190_constThr.py [hunk 3]
```diff
@@ -51,14 +51,14 @@ def main():
     
     # load older results
     #save_results(results)
-    old_results = load_results()
```
Comment (aerialhedgehog) [inline scripts/regression/test_mission_Embraer_E190_constThr.py]: check that these are still being used

File: scripts/regression/test_mission_B737.py [hunk 4]
```diff
@@ -50,14 +50,14 @@ def main():
 
     # load older results
     #save_results(results)
-    old_results = load_results()
```
Comment (aerialhedgehog) [inline scripts/regression/test_mission_B737.py]: check these yo

File: scripts/regression/test_solar_network.py [hunk 5]
```diff
@@ -119,14 +119,19 @@ def main():
     net.solar_logic       = logic
     
     # Setup the conditions to run the network
-    conditions                 = Data()
+    state = Data()
```
Comment (aerialhedgehog) [inline scripts/regression/test_solar_network.py]: how about using SUAVE.Analysis.Mission.Segments.Conditions.State()

File: scripts/regression/test_take_off_field_length.py [hunk 6]
```diff
@@ -444,7 +444,7 @@ def vehicle_setup():
     
     # add turbofan to vehicle
     vehicle.propulsors.append(turbofan)
-    
+    vehicle.network=turbofan
```
Comment (aerialhedgehog) [inline scripts/regression/test_take_off_field_length.py]: do you need it this?

File: scripts/tutorials/tut_opt_Embraer_E190_bat.py [hunk 7]
```diff
@@ -0,0 +1,1273 @@
...
+
+from pint import UnitRegistry
+from SUAVE.Core import Units as Units
+# ----------------------------------------------------------------------
+#   Inputs
+# ----------------------------------------------------------------------
+
+def main():
+    #use global variables for guess to make optimization faster
+    '''
```
Comment (aerialhedgehog) [inline scripts/tutorials/tut_opt_Embraer_E190_bat.py]: pretend these never existed

File: trunk/SUAVE/Analyses/Energy/Energy.py [hunk 8]
```diff
@@ -16,10 +16,10 @@ class Energy(Analysis):
     """
     def __defaults__(self):
         self.tag    = 'energy'
-        self.features = Data()
-        self.settings = Data()
         
+    def evaluate(self,state):
```
Comment (aerialhedgehog) [inline trunk/SUAVE/Analyses/Energy/Energy.py]: should be 
```
def evaluate_thrust()
```

File: trunk/SUAVE/Analyses/Energy/Energy.py [hunk 9]
```diff
@@ -16,10 +16,10 @@ class Energy(Analysis):
     """
     def __defaults__(self):
         self.tag    = 'energy'
-        self.features = Data()
-        self.settings = Data()
         
+    def evaluate(self,state):
+        network    = self.network
+        F, mdot, P = network.evaluate_thrust(state)
```
Comment (aerialhedgehog) [inline trunk/SUAVE/Analyses/Energy/Energy.py]: modifcation:
```
[ F, mdot, P ]
```

to 

```
results = Results()
results.thrust_force_vector = array([n x 3])
results.vehicle_mass_rate = array([n x 1])
```

File: trunk/SUAVE/Components/Propulsors/Propulsor.py [hunk 10]
```diff
@@ -58,71 +58,12 @@ class Container(Physical_Component.Container):
             example: find_instances(Propulsors.Turbojet) > return all Turbojets
```
Comment (aerialhedgehog) [inline trunk/SUAVE/Components/Propulsors/Propulsor.py]: remove power_flow and get_area

File: trunk/SUAVE/Methods/Missions/Segments/Common/Energy.py [hunk 11]
```diff
@@ -21,6 +21,36 @@ def initialize_battery(segment,state):
...
+        Assumptions -
+
+
+    """    
     
+    # unpack
+    energy_model = segment.analyses.energy
+
+    # evaluate
+    F, mdot, P   = energy_model(state)
```
Comment (aerialhedgehog) [inline trunk/SUAVE/Methods/Missions/Segments/Common/Energy.py]: use the results data structure here, call energy_model.evaluate_thrust()

File: scripts/experimental/Al_air_opt/full_setup.py [hunk 12]
```diff
@@ -0,0 +1,320 @@
...
+    # ------------------------------------------------------------------
+    #  Stability Analysis
+    stability = SUAVE.Analyses.Stability.Fidelity_Zero()
+    stability.geometry = vehicle
+    analyses.append(stability)
+    
+    # ------------------------------------------------------------------
+    #  Energy Analysis
+    energy= SUAVE.Analyses.Energy.Energy()
+    energy.network = vehicle.network #what is called throughout the mission (at every time step))
```
Comment (aerialhedgehog) [inline scripts/experimental/Al_air_opt/full_setup.py]: -> vehicle.energy_network

File: scripts/regression/test_take_off_field_length.py (hunk 6 above)
Comment (planes) [inline scripts/regression/test_take_off_field_length.py]: Nope you don't!

File: trunk/SUAVE/Analyses/Energy/Energy.py (hunk 8 above)
Comment (planes) [inline trunk/SUAVE/Analyses/Energy/Energy.py]: Done.

File: trunk/SUAVE/Components/Propulsors/Propulsor.py (hunk 10 above)
Comment (planes) [inline trunk/SUAVE/Components/Propulsors/Propulsor.py]: Done

**Pull request merged**